### PR TITLE
[test][easy] Add debug utils for cpu select algorithm test

### PR DESCRIPTION
**Pull request by henrylhtsang**
Summary: Add debug utils to debug a flaky test in fbcode ci.

Some context: https://github.com/pytorch/pytorch/pull/126545

Test Plan: ci

Differential Revision: D62005445

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by henrylhtsang** [comment]
@jgong5 can you take a look?

File: test/inductor/test_cpu_select_algorithm.py [hunk 1]
```diff
@@ -235,6 +239,16 @@ def __init__(self, bias, epilogue, other):
...
 
+        # TODO: debug utils, safe to remove in Oct 2024
+        log.warning(
+            f"DEBUG: torch.backends.mkl.is_available() is {torch.backends.mkl.is_available()}, "  # noqa: G004
+            f"torch.ops.mkldnn._is_mkldnn_fp16_supported() is {torch.ops.mkldnn._is_mkldnn_fp16_supported()}, "
+            f"torch.ops.mkldnn._is_mkldnn_bf16_supported() is {torch.ops.mkldnn._is_mkldnn_bf16_supported()}, "
+            f"inductor_config.freezing is {inductor_config.freezing}, "
+            f"mkldnn._is_mkldnn_acl_supported() is {torch.ops.mkldnn._is_mkldnn_acl_supported()}, "
+            f"torch._C.has_mkl is {torch._C.has_mkl}."
+        )
```
Comment (jgong5) [inline test/inductor/test_cpu_select_algorithm.py:250]: Since it is only used for debugging fbcode thing, can we wrap it with `is_fbcode()` check?

**Review by jgong5 — APPROVED** [review]
LGTM. Better to wrap it with is_fbcode() check though.

**Review by XuehaiPan — APPROVED** [review]

**Comment by henrylhtsang** [comment]
It looks like in some cases we have `torch.ops.mkldnn._is_mkldnn_bf16_supported() = False`. I don't know why that would happen occasionally, considering they even use the same hardware. 

Quickest way forward might be to add that check in the if conditions.

**Comment by henrylhtsang** [comment]
> It looks like in some cases we have `torch.ops.mkldnn._is_mkldnn_bf16_supported() = False`. I don't know why that would happen occasionally, considering they even use the same hardware.
> 
> Quickest way forward might be to add that check in the if conditions.

Correction: 
The `is_mkldnn_bf16_supported = False` one is using Intel(R) Xeon(R) CPU E5-2680 v4, while the `is_mkldnn_bf16_supported = True` one is using Intel(R) Xeon(R) Platinum 8321HC. I figured this might be something obvious, so @jgong5 do let me know if this is expected.

**Pull request closed without merging**